rust/src/dcerpc/dcerpc_udp.rs

```diff
@@ -67,37 +68,53 @@ pub struct DCERPCUDPState {
 impl DCERPCUDPState {
     pub fn new() -> DCERPCUDPState {
         return DCERPCUDPState {
+            tx_id: 0,
             header: None,
-            request: None,
-            response: None,
+            transactions: Vec::new(),
             fraglenleft: 0,
             uuid_entry: None,
             uuid_list: Vec::new(),
             de_state: None,
         };
     }
 
-    fn new_request(&mut self) {
-        let request = DCERPCRequest::new();
-        self.request = Some(request);
+    fn create_tx(&mut self, serial_no: u16) -> DCERPCTransaction {
+        let mut tx = DCERPCTransaction::new();
+        let endianness = self.get_hdr_drep_0() & 0x10;
+        tx.id = self.tx_id;
+        tx.call_id = serial_no as u32;
+        tx.endianness = endianness;
+        self.tx_id += 1;
+        tx
     }
 
-    fn new_response(&mut self) {
-        let response = DCERPCResponse::new();
-        self.response = Some(response);
+
+    fn evaluate_serial_no(&mut self) -> u16 {
+        let serial_no: u16;
+        let mut serial_hi: u8 = 0;
+        let mut serial_lo: u8 = 0;
+        let endianness = self.get_hdr_drep_0();
+        if let Some(ref hdr) = &self.header {
+            serial_hi = hdr.serial_hi;
+            serial_lo = hdr.serial_lo;
+        }
+        let mut cursor_serial_no = Cursor::new(vec![serial_hi, serial_lo]);
```

Comment:
@jasonish any thoughts on this `Cursor` use? See also https://github.com/OISF/suricata/pull/5036#discussion_r436698822

Reply:
Here it would seem simpler to to just use `serial_no = serial_hi << 8 | serial_lo` or vice-versa I think.  The cursor is more beneficial when you have to read out multiple values from the buffer I think.